File: 1337code/53.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int prev, maxVal;
        prev = maxVal = nums[0];

        for (int i = 1; i < nums.size(); i++) {
            int curr = max(prev + nums[i], nums[i]);
            maxVal = max(maxVal, curr);
            prev = curr;
        }

        return maxVal;
    }
};

int main() {
    cin.sync_with_stdio(0); cin.tie(0);
    freopen("53.input", "r", stdin); // For testing. Comment out for submissions

    int n; cin >> n;

    vector<int> nums(n);

    for (int i = 0; i < n; i++) {
        cin >> nums[i];
    }

    Solution s;
    cout << s.maxSubArray(nums) << endl;

    return 0;
}

File: 1337code/53.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSub = nums[0]
        currSub = nums[0]

        for num in range(1, len(nums)):
            currSub = max(currSub, 0)
            currSub += nums[num]
            maxSub = max(currSub, maxSub)
        
        return maxSub